src/goto-diff/unified_diff.cpp

```diff
@@ -412,8 +411,7 @@ bool unified_difft::instructions_equal(
     (ins1.targets.empty() ||
      instructions_equal(
        *ins1.get_target(),
-       *ins2.get_target(),
-       false));
```

Comment:
This is actually a bug (my bad): it should be used as `ins1.targets.empty() || !recurse || instructions_equal(...`

Reply:
I've added a commit which fixes this.